astroid/tests/unittest_brain.py

```diff
@@ -921,5 +918,134 @@ def test_subprocess_args(self):
+
+    def test_isinstance_object_true(self):
+        class Bar(object):
+            pass
+        assert _get_result("""
+        class Bar(object):
+            pass
+        isinstance(Bar(), object)
+        """) == "True" == str(isinstance(Bar(), object))
+
+    def test_isinstance_object_true3(self):
+        class Bar(object):
+            pass
+        assert _get_result("""
+        class Bar(object):
+            pass
+        isinstance(Bar(), Bar)
+        """) == "True" == str(isinstance(Bar(), Bar))
+    def test_isinstance_class_false(self):
+        class Foo(object):
+            pass
+        class Bar(object):
+            pass
+        assert _get_result("""
+        class Foo(object):
+            pass
+        class Bar(object):
+            pass
+        isinstance(Bar(), Foo)
+        """) == "False" == str(isinstance(Bar(), Foo))
+    def test_isinstance_type_false(self):
+        class Bar(object):
+            pass
+        assert _get_result("""
+        class Bar(object):
+            pass
+        isinstance(Bar(), type)
+        """) == "False" == str(isinstance(Bar(), type))
+
+    def test_isinstance_str_true(self):
+        """Make sure isinstance can check bultin str types"""
+        assert _get_result("isinstance('a', str)") == "True" == str(isinstance('a', str))
+
+    def test_isinstance_str_false(self):
+        assert _get_result("isinstance(1, str)") == "False" == str(isinstance(1, str))
+
+    def test_isinstance_tuple_argument(self):
+        """obj just has to be an instance of ANY class/type on the right"""
+        assert _get_result("isinstance(1, (str, int))") == "True" == str(isinstance(1, (str, int)))
+
+    def test_isinstance_type_false2(self):
+        assert _get_result("""
+        isinstance(1, type)
+        """) == "False" == str(isinstance(1, type))
+
+    def test_isinstance_object_true2(self):
+        class Bar(type):
+            pass
+        mainbar = Bar("Bar", tuple(), {})
+        assert _get_result("""
+        class Bar(type):
+            pass
+        mainbar = Bar("Bar", tuple(), {})
+        isinstance(mainbar, object)
+        """) == "True" == str(isinstance(mainbar, object))
+
+    def test_isinstance_type_true(self):
+        class Bar(type):
+            pass
+        mainbar = Bar("Bar", tuple(), {})
+        assert _get_result("""
+        class Bar(type):
+            pass
+        mainbar = Bar("Bar", tuple(), {})
+        isinstance(mainbar, type)
+        """) == "True" == str(isinstance(mainbar, type))
+
+    def test_isinstance_edge_case(self):
+        """isinstance allows bad type short-circuting"""
+        assert _get_result("isinstance(1, (int, 1))") == "True" == str(isinstance(1, (int, 1)))
+
+    def test_uninferable_bad_type(self):
+        """The second argument must be a class or a tuple of classes"""
+        # Should I subclass
+        with pytest.raises(astroid.TypeInferenceError):
```

Comment:
Ditto here, these should get back InferenceError/Uninferable, not an astroid exception.